src/Aspire.Dashboard/Utils/FormatHelpers.cs

```diff
@@ -26,30 +32,43 @@ static string GetLongTimePatternWithMilliseconds()
     [GeneratedRegex("(:ss|:s)")]
     private static partial Regex MatchSecondsInTimeFormatPattern();
 
-    private static readonly string s_longTimePatternWithMilliseconds = GetLongTimePatternWithMilliseconds();
-    private static readonly string s_shortDateLongTimePatternWithMilliseconds = DateTimeFormatInfo.CurrentInfo.ShortDatePattern + " " + s_longTimePatternWithMilliseconds;
+    private static MillisecondFormatStrings GetMillisecondFormatStrings(CultureInfo cultureInfo)
+    {
+        return s_formatStrings.GetOrAdd(cultureInfo.Name, static name =>
+        {
+            var c = CultureInfo.GetCultureInfo(name);
+            var longTimePatternWithMilliseconds = GetLongTimePatternWithMillisecondsCore(c);
```

Comment:
Consider making `GetLongTimePatternWithMillisecondsCore` a local function of this method, to ensure callers go through the cache.

Alternatively, make the access modifier on `GetLongTimePatternWithMillisecondsCore` explicit.